utils/NoiseGate.h

```diff
@@ -0,0 +1,135 @@
+#pragma once
+#ifndef DSY_NOISE_GATE_H
+#define DSY_NOISE_GATE_H
+
+#include <stdlib.h>
+#include "EnvelopeDetector.h"
+#include "AREnvelope.h"
+
+// TODO: until I'm able to get cmake to correctly import DaisySP,
+// I'm doing this hack to mock out dsp methods for the test environment.
+#ifdef DSYSP_H
+#include "Utility/dsp.h"
+
+#define POW_10F(v) daisysp::pow10f(v)
```

Comment:
I'm sure there's some CMake settings to make this unnecessary but I'm not great with C/C++ tooling